wrangler-service/src/main/java/co/cask/wrangler/service/bigquery/BigQueryService.java

```diff
@@ -0,0 +1,347 @@
+import co.cask.cdap.api.service.http.AbstractHttpServiceHandler;
+import co.cask.cdap.api.service.http.HttpServiceContext;
+import co.cask.cdap.api.service.http.HttpServiceRequest;
+import co.cask.cdap.api.service.http.HttpServiceResponder;
+import co.cask.wrangler.DataPrep;
+import co.cask.wrangler.PropertyIds;
+import co.cask.wrangler.RequestExtractor;
+import co.cask.wrangler.SamplingMethod;
+import co.cask.wrangler.ServiceUtils;
+import co.cask.wrangler.api.Row;
+import co.cask.wrangler.dataset.connections.Connection;
+import co.cask.wrangler.dataset.connections.ConnectionStore;
+import co.cask.wrangler.dataset.workspace.DataType;
+import co.cask.wrangler.dataset.workspace.WorkspaceDataset;
+import co.cask.wrangler.service.connections.ConnectionType;
+import co.cask.wrangler.service.gcp.GCPServiceAccount;
+import co.cask.wrangler.utils.ObjectSerDe;
+import com.google.api.gax.paging.Page;
+import com.google.api.services.bigquery.model.QueryRequest;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobId;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryResponse;
+import com.google.cloud.bigquery.QueryResult;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.storage.Storage;
+import com.google.common.base.Charsets;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+import static co.cask.wrangler.ServiceUtils.error;
+import static co.cask.wrangler.ServiceUtils.sendJson;
+import static co.cask.wrangler.service.directive.DirectivesService.WORKSPACE_DATASET;
+
+
+public class BigQueryService extends AbstractHttpServiceHandler {
+  private static final String PROJECT_ID = "projectId";
+  private static final String DATASET_ID = "datasetId";
+  private static final String TABLE_ID = "tableId";
+  private static final String SERVICE_ACCOUNT_KEYFILE = "service-account-keyfile";
+  private ConnectionStore store;
+
+  @UseDataSet(DataPrep.CONNECTIONS_DATASET)
+  private Table connectionTable;
+
+  @UseDataSet(WORKSPACE_DATASET)
+  private WorkspaceDataset ws;
+
+  @Override
+  public void initialize(HttpServiceContext context) throws Exception {
```

Comment:
I think we move this functionality into a common place to be used by GCS and BigQuery plus others that we add.